src/plasma/plasma_manager.cc

```diff
@@ -1446,12 +1439,9 @@ ClientConnection *ClientConnection_init(PlasmaManagerState *state,
                                         int client_sock,
                                         const char *client_key) {
   /* Create a new data connection context per client. */
-  ClientConnection *conn =
-      (ClientConnection *) malloc(sizeof(ClientConnection));
+  ClientConnection *conn = new ClientConnection();
```

Comment:
@pschafhalter I believe this was a bug. We forgot to replace the correspodning `free` call with a `delete` (in `ClientConnection_free`).